deployment-template/ec2/data-prepper-ec2-deployment-cfn.yaml

```diff
@@ -70,43 +109,84 @@ Resources:
+                      otel_trace_source:
+                        ssl: false
+                        health_check_service: true
+                    sink:
+                      - pipeline:
+                          name: "raw-pipeline"
+                      - pipeline:
+                          name: "service-map-pipeline"
+                  raw-pipeline:
+                    source:
+                      pipeline:
+                        name: "entry-pipeline"
+                    prepper:
+                      - otel_trace_raw_prepper:
+                    sink:
+                      - elasticsearch:
+                          hosts: [ "${AWSElasticsearchEndpoint}" ]
+                          aws_sigv4: true
+                          aws_region: "${AWSElasticsearchRegion}"
+                          trace_analytics_raw: true
+                  service-map-pipeline:
+                    delay: "100"
+                    source:
+                      pipeline:
+                        name: "entry-pipeline"
+                    prepper:
+                      - service_map_stateful:
+                    sink:
+                      - elasticsearch:
+                          hosts: [ "${AWSElasticsearchEndpoint}" ]
+                          aws_sigv4: true
+                          aws_region: "${AWSElasticsearchRegion}"
+                          trace_analytics_service_map: true
+                - !Sub |
+                  entry-pipeline:
+                    delay: "100"
+                    source:
+                      otel_trace_source:
+                        ssl: false
+                        health_check_service: true
+                    sink:
+                      - pipeline:
+                          name: "raw-pipeline"
+                      - pipeline:
+                          name: "service-map-pipeline"
+                  raw-pipeline:
+                    source:
+                      pipeline:
+                        name: "entry-pipeline"
+                    prepper:
+                      - otel_trace_raw_prepper:
+                    sink:
+                      - elasticsearch:
+                          hosts: [ "${AWSElasticsearchEndpoint}" ]
+                          aws_sigv4: false
+                          username: "${Username}"
+                          password: "${Password}"
+                          trace_analytics_raw: true
+                  service-map-pipeline:
+                    delay: "100"
+                    source:
+                      pipeline:
+                        name: "entry-pipeline"
+                    prepper:
+                      - service_map_stateful:
+                    sink:
+                      - elasticsearch:
+                          hosts: [ "${AWSElasticsearchEndpoint}" ]
+                          aws_sigv4: false
+                          username: "${Username}"
+                          password: "${Password}"
```

Comment:
Is there a scenario where both sigv4 _and_ username/password would be required?

Reply:
AFAICT through experiment, Amazon Elasticsearch Domain can not support IAM roles and HTTP basic auth simultaneously. Thus if HTTP auth is configured, one needs to whitelist the IAM ARN in custom access policy. Otherwise there is no way one can connect to the domain.